allennlp/tests/modules/token_embedders/pretrained_transformer_mismatched_embedder_test.py

```diff
@@ -0,0 +1,54 @@
+import torch
+
+from allennlp.common import Params
+from allennlp.data import Token, Vocabulary
+from allennlp.data.batch import Batch
+from allennlp.data.fields import TextField
+from allennlp.data.instance import Instance
+from allennlp.data.token_indexers import PretrainedTransformerMismatchedIndexer
+from allennlp.modules.text_field_embedders import BasicTextFieldEmbedder
+from allennlp.common.testing import AllenNlpTestCase
+
+
+class TestPretrainedTransformerMismatchedEmbedder(AllenNlpTestCase):
+    def test_end_to_end(self):
```

Comment:
This is a good test. One issue, however, is that given that it's using BERT we don't really have interpretable embeddings to make assertions against. So the logic in `forward` itself isn't really tested. One way around this would be to add an extra test and parameterize things such that you can pass in a fake `PretrainedTransformerEmbedder` that would return simple embeddings. This would be more work and I won't say it's strictly required, but if there happens to be a bug, we're going to catch it much later (and at greater cost) when our embeddings aren't giving us the performance we expect. So it's a tradeoff. I leave it to your discretion!

Reply:
I think one test that we can add similar to this purpose is when I verify that all this works in a real model, I can assert the output logits for a fixed input (assuming fixed seed, etc.). This helps prevent following changes accidentally breaking it.

Reply:
Yes, this is an option too. The benefit is that it's easy to implement. The downside is that it's brittle (an updated model from huggingface is liable to break it) and it's not interpretable, i.e., it won't necessarily be clear that the test shows that the embedder is working.

Reply:
Yeah, I would advise against hard-coding specific output embeddings or logits in a test, as that's really brittle.  Normally we haven't tried to test the specific predictions out of `forward` methods, as they are hard to test, we just test that it doesn't crash, and we rely on real-world usage to find those kinds of problems.  It's just too hard to catch in a unit test.